torch/export/__init__.py

```diff
@@ -314,12 +314,19 @@ def load(
     )
 
 
-def register_dataclass(cls: Type[Any]) -> None:
+def register_dataclass(
+    cls: Type[Any],
+    *,
+    serialized_type_name: Optional[str] = None,
```

Comment:
Should this default to None? Seems like if you register a dataclass for with export without setting this, and then you try to export it, it will just fail immediately, right?

Reply:
I don't think so? We just pass the None to torch.utils._pytree._register_pytree_node, and if it sees it's None then it'll set the type name to NO_SERIALIZED_TYPE_NAME_FOUND. It should only error if we try serializing.

Reply:
Sorry yeah, that's what I mean. If you export + serialize you are going to get an error. So I'm not sure what the valid reason for *not* setting this is?

Reply:
I was thinking, some people might just want to export but not serialize, just play with export a little bit, or reuse this function for some other pytree thing?

Reply:
hmmm, seems weird but sure lol